yt/sample_data/api.py

```diff
@@ -0,0 +1,176 @@
+import json
+import re
+from itertools import chain
+from pathlib import Path
+
+import pandas as pd
+import pkg_resources
+import pooch
+import requests
+
+from yt.funcs import mylog
+
+num_exp = re.compile(r"[0-9]+")
+byte_unit_exp = re.compile(r"[KMG]?B")
+
+
+def _parse_byte_size(s):
+    """
+    Convert a string size specification to integer byte size.
+
+    This function should be insensitive to case and whitespace.
+    Examples
+    --------
+    >>> parse_byte_size("11B")
+    11
+    >>> parse_byte_size("10 kB")
+    10000
+    >>> parse_byte_size("67.4GB")
+    67400000000
+    """
+    try:
+        s = s.upper()
+    except AttributeError:
+        # input is not a string (likely a np.nan)
+        return pd.NA
+    val = float(re.search(num_exp, s).group())
+    unit = re.search(byte_unit_exp, s).group()
+    exponent = {"B": 0, "KB": 3, "MB": 6, "GB": 9}[unit]
+    return int(val * 10 ** exponent)
```

Comment:
Should this be instead
```suggestion
    exponent = {"B": 0, "KB": 10, "MB": 20, "GB": 30}[unit]
    return int(val * 2 ** exponent)
```
? That's actually a fair question, I don't know if the sizes in the registry are in metric or base-2...

Reply:
Wait do you mean Kilo Mega and Giga are ever used outside base ten ??

Reply:
Well...  `ls` reports `94848B` as `93KB`, and not `95KB` :D
![image](https://user-images.githubusercontent.com/5411875/97454307-e5d80500-1936-11eb-9a01-127ffb289413.png)

Reply:
wow. I don't think we can safely assume those values were reliably reported so... Maybe I could just normalise the corresponding column in a companion PR, but that would require downloading everything to my drive, and I can't afford this atm.

Reply:
I think that the two are used interchangeably (they should not !), but the difference for 1GiB vs. 1GB is 7% "only". I am fine with you leaving it as is.